shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -215,30 +216,54 @@ private void setSystemChromeApplicationSwitcherDescription(
     }
   }
 
-  private void setSystemChromeChangeListener() {
-    // Set up a listener to notify the framework when the system ui has changed.
+  private void setSystemChromeChangeListenerLegacy() {
```

Comment:
This wasn't tested before and I believe that is because accessing the listener in a test is difficult to do with mocking. @Piinks is that right? 

I already tested this PR in FROB to confirm the behavior remains consistent.

Reply:
There might be a way to test it, you might be able to set or listen to the method `platformChannel.systemChromeChanged` and test that it calls with the right boolean based on the current fullscreen mode.

Reply:
@Piinks  I tried this but setting the visibility of the system bars with the mock was not getting picked up by the listener :/ Do you have any examples of testing a listener like this? The one I referred to earlier was [this](https://github.com/flutter/engine/blob/c058dd1896010a03f2dc60056d52b66de0faab26/shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java#L2030).

Reply:
Can you include this test so I can see it?